Hddm python mods

rjones30 opened the pull request:
- programs/Utilities/hddm/hddm-py.cpp [rtj]
  - several miscellaneous fixes to th recently added python hddm
    modules
- programs/Utilities/hddm/exam2.xml
  - rename attribute "pass" to "Pass" in the example hddm template to
    avoid collision with the python keyword "pass".

rjones30 committed:
* SBMS/sbms.py [rtj]
   - fix a mis-spelling in the python_so_module function that left an
     unitialized variable called mylib, renamed to the correct name.

rjones30 committed:
* programs/Utilities/hddm/hddm-py.cpp [rtj]
   - several miscellaneous fixes to th recently added python hddm
     modules

* programs/Utilities/hddm/exam2.xml
   - rename attribute "pass" to "Pass" in the example hddm template to
     avoid collision with the python keyword "pass".

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods

rjones30 committed:
* programs/Utilities/hddm/hddm-py.cpp [rtj]
   - was missing a terminating semicolon on one line

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods

rjones30 committed:
* programs/Utilities/hddm/hddm-py.cpp [rtj]
   - modified so the setup script knows where to find the source files
     that are needed for the build when then build takes place outside
     the source directory.
* libraries/HDDM/SConscript [rtj]
   - add a dependency for the c++ implementation source file when
     building the python module, because static data members defined
     there and are needed to build the module.

gluex commented:
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-logs)

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods

rjones30 committed:
* hddm-c.cpp [rtj]
   - add a cast for a char[] string initializer to suppress
     compiler warnings.

gluex commented:
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-logs)

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods
